### PR TITLE
refactor: LEAP-1300: Remove outdated FF_LSDV_4620_3

### PR DESCRIPTION
**Pull request by Gondragos**
### Reason for change
The feature flag `fflag_feat_front_lsdv_4620_richtext_opimization_060423_short` and its associated legacy code were no longer in use. This removes outdated flags, unused styles, and unnecessary tests.

A small refactoring has also been done there

| **Key** | `fflag_feat_front_lsdv_4620_richtext_opimization_060423_short` |
|---|---|
| **Variable Name** | `FF_LSDV_4620_3` |
| **Ends up as** | 🟩 `True` |
| **Description** | Reworking of RichText to optimize its work with DOM and decrease response time with a large number of regions. It also fixes scenarios of working with hidden regions and edge cases for creating spans inside other spans. |

### Rollout strategy
This change removes already-unused code and does not require rollout configurations.

### Testing
- Verified that all occurrences of the feature flag and its related codebase references have been successfully removed.
- Ensured no regression by running existing test suites.

### Risks

- No significant risks are identified as the feature flag and related code are not in use.
- Minimal potential for unforeseen regressions due to cleanup of legacy code.

### Reviewer notes
- I did not found where `regsObserverDisposer` is initialized or used, so I've removed it. I'm not sure there is no some black magic in code. [[1]](https://github.com/HumanSignal/label-studio/pull/7228/files?diff=unified&w=1#diff-e71e84686ed84eb6be7fc89c28a61fa93fb496c7d8a4b5890d18471b7f3e3f34L268) [[2]](https://github.com/HumanSignal/label-studio/pull/7228/files?diff=unified&w=1#diff-e71e84686ed84eb6be7fc89c28a61fa93fb496c7d8a4b5890d18471b7f3e3f34L160)
- The other part there is a reason to check precisely is using of new getters [[3]](https://github.com/HumanSignal/label-studio/pull/7228/files#diff-e71e84686ed84eb6be7fc89c28a61fa93fb496c7d8a4b5890d18471b7f3e3f34R147-R157)
- Please review the removal of the feature flag and ensure all associated legacy implementations, tests, and styles are removed correctly

### General notes
This change focuses solely on cleaning up unused functionality and should not impact existing feature behavior.

### CODE REVIEW

File: web/libs/editor/tests/e2e/tests/rich-text/rich-text-edge-cases.test.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const edgeCaseConfig = `<View>
     <HyperText name="html" value="$html" />
 </View>`;
 const edgeCaseHtml = `<p>This
-is an <u><span>e</span><span>x</span><span>a</span><span>m</span><span>p</span><span>l</span><span>e</span></u><br/> of 
+is an <u><span>e</span><span>x</span><span>a</span><span>m</span><span>p</span><span>l</span><span>e</span></u><br/> of
```
Comment (Gondragos) [inline web/libs/editor/tests/e2e/tests/rich-text/rich-text-edge-cases.test.js]: That linting keeps killing test data.
```suggestion
is an <u><span>e</span><span>x</span><span>a</span><span>m</span><span>p</span><span>l</span><span>e</span></u><br/> of 
```

**Comment by Gondragos** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13900734030)
> Successfully merged:  3 files changed, 31 insertions(+), 4 deletions(-)

**Review by nass600 — APPROVED** [review]

File: web/libs/editor/src/mixins/HighlightMixin.js [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
 const HIGHLIGHT_NO_LABEL_CN = "htx-no-label";
-const IDENTIFIER_LENGTH = 5;
 const LABEL_COLOR_ALPHA = 0.3;
 
 export const HighlightMixin = types
   .model()
   .views((self) => ({
     get _hasSpans() {
       // @todo is it possible that only some spans are connected?
+      // @TODO: Is it a legacy code we no need anymore?
```
Comment (Gondragos) [inline web/libs/editor/src/mixins/HighlightMixin.js]:
```suggestion
      // @TODO: Need to check if it is still necessary condition. The way of working with spans was changed and it could affect this part. The main question, is there still a way to get `isConnected === false`
```

File: web/libs/editor/src/tags/object/RichText/model.js [hunk 3]
```diff
@@ -144,33 +144,26 @@ const Model = types
       }
       `;
     },
+    getIframeBodyNode() {
```
Comment (Gondragos) [inline web/libs/editor/src/tags/object/RichText/model.js:148]:
```suggestion
    // This is not a real getter as it is dependant on ref wich cannot be cached in the right way
    getIframeBodyNode() {
```

File: web/libs/editor/src/tags/object/RichText/model.js [hunk 4]
```diff
@@ -144,33 +144,26 @@ const Model = types
...
+      const mountNode = self.mountNodeRef.current;
+      return mountNode?.contentDocument?.body;
+    },
+    getRootNode() {
```
Comment (Gondragos) [inline web/libs/editor/src/tags/object/RichText/model.js:153]:
```suggestion
    // This is not a real getter as it is dependant on ref wich cannot be cached in the right way
    getRootNode() {
```

File: web/libs/editor/src/tags/object/RichText/model.js [hunk 5]
```diff
@@ -144,33 +144,28 @@ const Model = types
       }
       `;
     },
+    // This is not a real getter as it is dependant on ref wich cannot be cached in the right way
```
Comment (Gondragos) [inline web/libs/editor/src/tags/object/RichText/model.js]:
```suggestion
    // This is not a real getter as it is dependant on ref which cannot be cached in the right way
```

File: web/libs/editor/src/tags/object/RichText/model.js [hunk 6]
```diff
@@ -144,33 +144,28 @@ const Model = types
...
+    getIframeBodyNode() {
+      const mountNode = self.mountNodeRef.current;
+      return mountNode?.contentDocument?.body;
+    },
+    // This is not a real getter as it is dependant on ref wich cannot be cached in the right way
```
Comment (Gondragos) [inline web/libs/editor/src/tags/object/RichText/model.js]:
```suggestion
    // This is not a real getter as it is dependant on ref which cannot be cached in the right way
```

**Review by hlomzik — APPROVED** [review]

**Comment by hlomzik** [comment]
That's a long and complex PR, wow! Thank you for fixing everything and clarifying difficult moments. Well-deserved approve!

**Comment by Gondragos** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13927384305)
> Successfully merged:  delete mode 100644 web/libs/datamanager/src/assets/icons/refresh2.svg
